-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49021/46266

ERROR: Build errors found during clang-tidy run.
```
src/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/PixelClustering.h:181:1: error: version control conflict marker in file [clang-diagnostic-error]
  181 | <<<<<<< HEAD
      | ^
Suppressed 690 warnings (690 in non-user code).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/Geometry/CommonTopologies/interface/SimplePixelTopology.h
+++ b/Geometry/CommonTopologies/interface/SimplePixelTopology.h
@@ -377,7 +377,7 @@ namespace pixelTopology {
     static constexpr uint16_t last_barrel_detIndex = 864;
 
     static constexpr uint32_t maxPixInModule = 6000;
-    static constexpr uint32_t maxPixInModuleForMorphing = maxPixInModule;
+    static constexpr uint32_t maxPixInModuleForMorphing = 0;
     static constexpr uint32_t maxIterClustering = 16;
 
     static constexpr uint32_t maxNumClustersPerModules = phase2PixelTopology::maxNumClustersPerModules;
@@ -474,7 +474,7 @@ namespace pixelTopology {
     static constexpr uint16_t last_barrel_detIndex = 1184;
 
     static constexpr uint32_t maxPixInModule = 6000;
-    static constexpr uint32_t maxPixInModuleForMorphing = maxPixInModule * 7/5;
+    static constexpr uint32_t maxPixInModuleForMorphing = maxPixInModule * 2 / 5;
     static constexpr uint32_t maxIterClustering = 24;
 
     static constexpr uint32_t maxNumClustersPerModules = phase1PixelTopology::maxNumClustersPerModules;
@@ -588,7 +588,7 @@ namespace pixelTopology {
     static constexpr uint32_t maxNumberOfQuadruplets = maxNumberOfTuples;
 
     static constexpr uint32_t maxPixInModule = 10000;
-    static constexpr uint32_t maxPixInModuleForMorphing = maxPixInModule * 11/10;
+    static constexpr uint32_t maxPixInModuleForMorphing = maxPixInModule * 1 / 10;
     static constexpr uint32_t maxIterClustering = 32;
 
     static constexpr uint32_t maxNumOfActiveDoublets =

--- a/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/PixelClustering.h
+++ b/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/PixelClustering.h
@@ -17,7 +17,7 @@
 #include "HeterogeneousCore/AlpakaInterface/interface/config.h"
 #include "HeterogeneousCore/AlpakaInterface/interface/warpsize.h"
 
-// #define GPU_DEBUG
+//#define GPU_DEBUG
 
 // TODO move to HeterogeneousCore/AlpakaInterface or upstream to alpaka
 template <typename TAcc, typename T>
@@ -174,14 +174,15 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::pixelClustering {
   template <typename TrackerTraits>
   struct FindClus {
     // assume that we can cover the whole module with up to 16 blockDimension-wide iterations
-  
+
     // this must be larger than maxPixInModule / maxIterClustering, and should be a multiple of the warp size
     static constexpr uint32_t maxElementsPerBlock =
         cms::alpakatools::round_up_by(TrackerTraits::maxPixInModule / TrackerTraits::maxIterClustering, 64);
-    static constexpr uint32_t maxElementsPerBlockMorph =
-        cms::alpakatools::round_up_by(TrackerTraits::maxPixInModuleForMorphing / TrackerTraits::maxIterClustering, 64);
-    static_assert(maxElementsPerBlockMorph>=maxElementsPerBlock);
-    
+    static constexpr uint32_t maxElementsPerBlockMorph = cms::alpakatools::round_up_by(
+        (TrackerTraits::maxPixInModule + TrackerTraits::maxPixInModuleForMorphing) / TrackerTraits::maxIterClustering,
+        64);
+    static_assert(maxElementsPerBlockMorph >= maxElementsPerBlock);
+
     ALPAKA_FN_ACC void operator()(Acc1D const& acc,
                                   SiPixelDigisSoAView digi_view,
                                   SiPixelDigisSoAView fakes_view,
@@ -261,11 +262,12 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::pixelClustering {
           }
         }
 
-        using Hist = cms::alpakatools::HistoContainer<uint16_t,
-                                                      TrackerTraits::clusterBinning,
-                                                      TrackerTraits::maxPixInModuleForMorphing,
-                                                      TrackerTraits::clusterBits,
-                                                      uint16_t>;
+        using Hist =
+            cms::alpakatools::HistoContainer<uint16_t,
+                                             TrackerTraits::clusterBinning,
+                                             TrackerTraits::maxPixInModule + TrackerTraits::maxPixInModuleForMorphing,
+                                             TrackerTraits::clusterBits,
+                                             uint16_t>;
         constexpr int warpSize = cms::alpakatools::warpSize;
         auto& hist = alpaka::declareSharedVar<Hist, __COUNTER__>(acc);
         auto& ws = alpaka::declareSharedVar<typename Hist::Counter[warpSize], __COUNTER__>(acc);
@@ -574,16 +576,16 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::pixelClustering {
         ALPAKA_ASSERT_ACC((hist.size() / blockDimension) < TrackerTraits::maxIterClustering);
 
         // number of elements per thread
-        const uint32_t maxElements =
-            cms::alpakatools::requires_single_thread_per_block_v<Acc1D> ?  (enableDigiMorphing ? maxElementsPerBlockMorph : maxElementsPerBlock) : 1;
-        
-       
+        const uint32_t maxElements = cms::alpakatools::requires_single_thread_per_block_v<Acc1D>
+                                         ? (enableDigiMorphing ? maxElementsPerBlockMorph : maxElementsPerBlock)
+                                         : 1;
+
 #ifdef GPU_DEBUG
-        const auto nthreads = alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u];
-        if (nthreads > maxElements)
-          printf("This is WRONG: nthreads > maxElements : %d > %d\n",nthreads,maxElements);
+        const auto nElementsPerThread = alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u];
+        if (nElementsPerThread > maxElements)
+          printf("This is WRONG: nElementsPerThread > maxElements : %d > %d\n", nElementsPerThread, maxElements);
         else if (thisModuleId % 500 == 1)
-          printf("This is OK: nthreads <= maxElements : %d <= %d\n",nthreads,maxElements);
+          printf("This is OK: nElementsPerThread <= maxElements : %d <= %d\n", nElementsPerThread, maxElements);
 #endif
 
         ALPAKA_ASSERT_ACC((alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u] <= maxElements));

--- a/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToCluster.cc
+++ b/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToCluster.cc
@@ -210,13 +210,14 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     digiMorphingConfig_.applyDigiMorphing = iConfig.getParameter<bool>("DoDigiMorphing");
     digiMorphingConfig_.maxFakesInModule = iConfig.getParameter<uint32_t>("MaxFakesInModule");
 
-    if (digiMorphingConfig_.maxFakesInModule > TrackerTraits::maxPixInModuleForMorphing - TrackerTraits::maxPixInModule) {
-      throw cms::Exception("Configuration") << "[SiPixelDigiMorphing]:"
-                                            << " maxFakesInModule should be <= " << TrackerTraits::maxPixInModuleForMorphing - TrackerTraits::maxPixInModule 
-                                            << " (TrackerTraits::maxPixInModuleForMorphing - TrackerTraits::maxPixInModule)"
-                                            << " while " << digiMorphingConfig_.maxFakesInModule << " was provided at config level.\n";
+    if (digiMorphingConfig_.maxFakesInModule > TrackerTraits::maxPixInModuleForMorphing) {
+      throw cms::Exception("Configuration")
+          << "[SiPixelDigiMorphing]:"
+          << " maxFakesInModule should be <= " << TrackerTraits::maxPixInModuleForMorphing
+          << " (TrackerTraits::maxPixInModuleForMorphing)"
+          << " while " << digiMorphingConfig_.maxFakesInModule << " was provided at config level.\n";
     }
-    
+
     // regions
     if (!iConfig.getParameter<edm::ParameterSet>("Regions").getParameterNames().empty()) {
       regions_ = std::make_unique<PixelUnpackingRegions>(iConfig, consumesCollector());
@@ -240,7 +241,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     desc.add<double>("VCaltoElectronOffset", -60.f);
     desc.add<double>("VCaltoElectronOffset_L1", -670.f);
     desc.add<bool>("DoDigiMorphing", false);
-    desc.add<uint32_t>("MaxFakesInModule", TrackerTraits::maxPixInModuleForMorphing - TrackerTraits::maxPixInModule);
+    desc.add<uint32_t>("MaxFakesInModule", TrackerTraits::maxPixInModuleForMorphing);
 
     desc.add<edm::InputTag>("InputLabel", edm::InputTag("rawDataCollector"));
     {

--- a/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc
+++ b/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc
@@ -291,7 +291,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     // Kernel to perform Raw to Digi conversion
     template <bool debug = false>
     struct RawToDigi_kernel {
-    ALPAKA_FN_ACC void operator()(Acc1D const &acc,
+      ALPAKA_FN_ACC void operator()(Acc1D const &acc,
                                     const SiPixelMappingSoAConstView &cablingMap,
                                     const unsigned char *modToUnp,
                                     const uint32_t wordCounter,
@@ -571,7 +571,9 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
         {
           const int blocks = 64;
 
-          const auto elementsPerBlockFindClus = digiMorphingConfig.applyDigiMorphing ? FindClus<TrackerTraits>::maxElementsPerBlockMorph : FindClus<TrackerTraits>::maxElementsPerBlock;
+          const auto elementsPerBlockFindClus = digiMorphingConfig.applyDigiMorphing
+                                                    ? FindClus<TrackerTraits>::maxElementsPerBlockMorph
+                                                    : FindClus<TrackerTraits>::maxElementsPerBlock;
           const auto workDivFindClus = cms::alpakatools::make_workdiv<Acc1D>(blocks, elementsPerBlockFindClus);
 
           // allocate a transient collection for the fake pixels recovered by the digi morphing algorithm
@@ -591,7 +593,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
                               digiMorphingConfig.applyDigiMorphing,
                               morphingModulesDevice,
                               digiMorphingConfig.numMorphingModules,
-                              digiMorphingConfig.maxFakesInModule, 
+                              digiMorphingConfig.maxFakesInModule,
                               clusters_d->view(),
                               wordCounter);
 #ifdef GPU_DEBUG
@@ -767,5 +769,5 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     template class SiPixelRawToClusterKernel<pixelTopology::HIonPhase1>;
 
   }  // namespace pixelDetails
-  
+
 }  // namespace ALPAKA_ACCELERATOR_NAMESPACE

